roles/overpass-api/tasks/compile.yml

```diff
@@ -0,0 +1,54 @@
+- name: install packages
+  apt:
+    pkg: "{{ item }}"
+  with_items:
+    - autoconf
+    - automake1.11
+    - expat
+    - git
+    - g++
+    - libtool
+    - libexpat1-dev
+    - make
+    - zlib1g-dev
+    - bzip2
+    - wget
+    - liblz4-1
+    - liblz4-dev
+
+- name: download {{ overpass_version }} release of overpass-api
+  unarchive:
+    src: "http://dev.overpass-api.de/releases/osm-3s_{{ overpass_version }}.tar.gz"
+    dest: "{{ project_dir }}"
+    remote_src: true
+  become: yes
+  become_user: "{{ user }}"
+
+- name: compile overpass-api
+  shell: |
+    ./configure --enable-lz4 CXXFLAGS="-O2" &&
+    make -j $(nproc --all)
+  args:
+    chdir: "{{ project_dir }}/osm-3s_{{ overpass_version }}"
+    creates: bin/dispatcher
+  become: yes
+  become_user: "{{ user }}"
+
+- name: install overpass-api executables
+  shell: make install
+  args:
+    chdir: "{{ project_dir }}/osm-3s_{{ overpass_version }}"
+
+- name: install overpass-api cgi-bin
+  command: cp -r {{ project_dir }}/osm-3s_{{ overpass_version }}/cgi-bin/ {{ overpass_webroot_dir }}/
```

Comment:
It could be done with ansible copy module with remote_src=yes, but it is fine as it is.